plugins/linux-pulseaudio/pulse-input.c

```diff
@@ -392,31 +397,20 @@ static obs_properties_t *pulse_output_properties(void *unused)
 /**
  * Server info callback
  */
-static void pulse_input_device(pa_context *c, const pa_server_info *i,
+static void pulse_default_devices(pa_context *c, const pa_server_info *i,
 	void *userdata)
 {
 	UNUSED_PARAMETER(c);
-	obs_data_t *settings = (obs_data_t*) userdata;
-
-	obs_data_set_default_string(settings, "device_id",
-		i->default_source_name);
-	blog(LOG_DEBUG, "Default input device: '%s'", i->default_source_name);
-
-	pulse_signal(0);
-}
+	struct pulse_devices *d = (struct pulse_devices*) userdata;
 
-static void pulse_output_device(pa_context *c, const pa_server_info *i,
-	void *userdata)
-{
-	UNUSED_PARAMETER(c);
-	obs_data_t *settings = (obs_data_t*) userdata;
+	d->default_source_name = i->default_source_name;
```

Comment:
You are creating a reference to a pulse audio string and you don't know how long is is valid after the end of the callback, you have to duplicate it, to make sure this does not cause any errors.